### PR TITLE
Update mongodb and bson to 2.0.0-beta.2

### PR DESCRIPTION
**Pull request by coniesco**

### CODE REVIEW

**Comment by dariusc93** [comment]
Quickly scanning over the changes on the mongodb crate and this PR, it looks like it might be better not to have `Document` used but instead pass the type where necessary so instead of doing something like `Collection<Document>`, one could do `Collection<T>` where T would be the type, which the developer could then either use `Document` or their own that implements `Serialize` and `Deserialize` from serde. I guess using `Document` would keep it backwards compatible though, but just my opinion. However, it is still a start. I havent had the chance to fully go over everything (which I may do in the morning) so I may have a couple of comments once I do go over the chances to mongodb crate and the pr you made. @alexkornitzer might have a different say in it though.

**Comment by alexkornitzer** [comment]
Thanks for having a scan :) I am all for breaking changes and this is why the crate is still in `0.x.0` numbers. 

So it looks like they have finally removed the need to pass everything around as a `Document`. My current thinking is that we should now bin the `Document` intermediate for collection types as this will allow a lot of derive boilerplate to be removed. This approach will also fix #2. So we would be going from:
```
Foo <-> Document <-> Bson
```
to
```
Foo <-> Bson
```
while keeping the document intermediates for things like `Filter`s etc.

On the ability to freely interchange between a collection's type and `Document`, I am not sure I like this idea, I am still very much in favour of keeping the type restrictions, i.e. you can't pass the wrong filter to the wrong type, but maybe there is a use case for this I just can't see?

@dariusc93 what are your additional thoughts?

File: mongod/src/async/client.rs [hunk 1]
```diff
@@ -301,10 +306,10 @@ impl ClientBuilder {
                             Some(tls_insecure.parse().map_err(crate::error::builder)?);
                     }
                     if let Some(ca_file_path) = self.ca.or(tls_ca_file) {
-                        options.ca_file_path = Some(ca_file_path);
+                        options.ca_file_path = Some(ca_file_path.parse().unwrap());
```
Comment (alexkornitzer) [inline mongod/src/async/client.rs:309]: We need to handle these unwraps, unwraps that can unwrap in prod are not good.

Comment (alexkornitzer) [inline mongod/src/async/client.rs:309]: We need to handle these unwraps, unwraps that can unwrap in prod are not good.

File: mongod/src/async/client.rs [hunk 2]
```diff
@@ -301,10 +306,10 @@ impl ClientBuilder {
...
                     }
                     if let Some(cert_key_file_path) = self.cert_key.or(tls_certificate_key_file) {
-                        options.cert_key_file_path = Some(cert_key_file_path);
+                        options.cert_key_file_path = Some(cert_key_file_path.parse().unwrap());
```
Comment (alexkornitzer) [inline mongod/src/async/client.rs:312]: Handle this unwrap too.

File: mongod/src/query/find.rs [hunk 3]
```diff
@@ -1,7 +1,7 @@
 use std::marker::PhantomData;
 use std::time::Duration;
 
-use bson::Document;
+use db::bson::Document;
```
Comment (alexkornitzer) [inline mongod/src/query/find.rs:4]: This should be `mongodb` not `db`. The `db` rename is for external use.

File: mongod/src/query/find.rs [hunk 4]
```diff
@@ -10,6 +10,8 @@ use crate::field::{AsField, Field};
 use crate::filter::{AsFilter, Filter};
 use crate::r#async::Client;
 use crate::sort::Sort;
+use serde::de::DeserializeOwned;
```
Comment (alexkornitzer) [inline mongod/src/query/find.rs:13]: I like to group imports together in the following order:
```
std

3rd party

local
```

Can we get these shunted up to line 6?

**Review by alexkornitzer — CHANGES_REQUESTED** [review]
Please can you address the listed requests.

**Comment by alexkornitzer** [comment]
Ah damn, if we remove the `Document` intermediate we are going to have to come up with a solution for `Document` updates which understandably still relies on `Document`. Maybe we should keep this part for updates only and bin it for the rest?

**Pull request merged**